cypress/integration/locks.js

```diff
@@ -1,8 +1,49 @@
 describe("Locks", function() {
-    // beforeEach(function() {
-    //     cy.getUserAEurBalance().as("startingAeurBalance");
-    //     cy.getUserEthBalance().as("startingEthBalance");
-    // });
+    const getLock = (prodId, disbursedAmount) => {
+        cy.get("[data-testid=lockMenuLink]").click();
 
-    it("Should lock and release A-EUR");
+        cy.get(`[data-testid=selectLockProduct-${prodId}]`).then(() => {
+            cy
+                .get("[data-testid=lockAmountInput]")
+                .type(disbursedAmount.toString())
+                .should("have.value", disbursedAmount.toString())
+        });
+        cy.get(`[data-testid=selectLockProduct-${prodId}]`).click();
+
+        cy.get(`[data-testid=submitButton]`).click();
+        cy.get("[data-testid=EthSubmissionSuccessPanel]").contains("New Lock submitted")
+        cy.get("[data-testid=EthSubmissionSuccessPanel] >[data-testid=msgPanelOkButton]").click();
+        
+        return cy.get("[data-testid=EthConfirmationReceivedPanel]").within(() => {
+            cy.contains("New lock");
+            cy.contains("Tx hash:");
+        });
+    };
+    
+    beforeEach(function() {
+        cy.getUserAEurBalance().as("startingAeurBalance");
+        cy.getUserEthBalance().as("startingEthBalance");
+    });
+    
+    it("Should lock A-EUR", function() {
+        getLock(7, 50).then(() => {
+            cy.assertUserAEurBalanceOnUI(this.startingAeurBalance - 50);
+            cy.get("[data-testid=EthConfirmationReceivedPanel] > [data-testid=msgPanelOkButton]").click();
+        });
+    });
+
+    it("Should release locked A-EUR", function() {        
+        getLock(7, 50).then(() => {
+            cy.get("[data-testid=EthConfirmationReceivedPanel]").within(() => {
+                cy.contains("12. confirmation", { timeout: 60000 });
+            });
+            cy.get("[data-testid=myAccountMenuLink]").click();
+            cy.wait(65000).then(() => {
+                cy.reload(true);
+                cy.wait(30000);
+                cy.get("[data-testid=LockListBlock] [data-testid=releaseLockButton]").first().click();
+                cy.get("[data-testid=EthSubmissionSuccessPanel]").contains("Funds release transaction submitted");
```

Comment:
I wouldn't over test this (as we should keep these  end-to-end test simple, happy path only) but as a minimum I would check user new user balance after lock and release too. See loan tests as an example